src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -175,6 +175,21 @@ AND NOT EXISTS (SELECT prs FROM p.results prs
             """)
     long countAssessmentsByExerciseIdSubmittedIgnoreTestRunSubmissions(@Param("exerciseId") Long exerciseId);
 
+    @Query("""
+
+            SELECT COUNT(DISTINCT p) FROM StudentParticipation p left join p.submissions s  left join s.results r
+            WHERE p.exercise.id = :exerciseId
+            AND s.id = (select max(id) from p.submissions)
+            AND r.assessor IS NOT NULL
+            AND r.rated = TRUE
+            AND r.submission.submitted = TRUE
```

Comment:
```suggestion
            AND s.submitted = TRUE
```

Reply:
not sure if this would work, but I don't see a reason to navigate from submission to result and then back to submission

Reply:
yeah, that's right, it should work. We'll try it. The r.submission.submitted comes from the query we used as a basis for ours. There wasn't a join to submission, so it was necessary before.